pkg/event/processor_test.go

```diff
@@ -449,3 +448,153 @@ func TestChanQueueEventProcessor_ProcessBatch(t *testing.T) {
 		assert.True(t, len(logEvent.Event.Visitors) >= 1)
 	}
 }
+
+type NoOpLogger struct {
+}
+
+
+func (l *NoOpLogger) Log(level logging.LogLevel, message string, fields map[string]interface{}) {
+}
+
+
+func (l *NoOpLogger) SetLogLevel(level logging.LogLevel) {
+
+}
+
+func BenchmarkWithQueueSize(b *testing.B) {
+	logging.SetLogger(&NoOpLogger{})
+
+	merges := []struct {
+		name string
+		fun  func(*testing.B)
+	}{
+		{"QueueSize100", benchmarkProcessor100},
+		{"QueueSize500", benchmarkProcessor500},
+		{"QueueSize1000", benchmarkProcessor1000},
+		{"QueueSize2000", benchmarkProcessor2000},
+		{"QueueSize3000", benchmarkProcessor3000},
+		{"QueueSize4000", benchmarkProcessor4000},
+	}
+
+	for _, merge := range merges {
+		b.Run(merge.name, merge.fun)
+	}
+}
+
+func BenchmarkWithBatchSize(b *testing.B) {
+	logging.SetLogger(&NoOpLogger{})
+
+	merges := []struct {
+		name string
+		batchSize int
+		fun  func(bs int, b *testing.B) int
+	}{
+		{"BatchSize10", 10, benchmarkProcessorWithBatchSize},
+		{"BatchSize20", 20,benchmarkProcessorWithBatchSize},
+		{"BatchSize30", 30, benchmarkProcessorWithBatchSize},
+		{"BatchSize40", 40,benchmarkProcessorWithBatchSize},
+		{"BatchSize50", 50, benchmarkProcessorWithBatchSize},
+		{"BatchSize60", 60, benchmarkProcessorWithBatchSize},
+	}
+
+	for _, merge := range merges {
+		b.Run(merge.name, func (b *testing.B) {
+			benchmarkProcessorWithBatchSize(merge.batchSize, b)
+		})
+	}
+
+}
+
+func BenchmarkWithQueue(b *testing.B) {
+	logging.SetLogger(&NoOpLogger{})
+
+	b.Run("InMemoryQueue", func (b *testing.B) {
+		benchmarkProcessorQueue(NewInMemoryQueue(defaultQueueSize), b)
+	})
+
+	b.Run("ChannelQueue", func (b *testing.B) {
+		benchmarkProcessorQueue(NewChanQueue(defaultQueueSize), b)
+	})
+
+}
+
+func benchmarkProcessor100(b *testing.B) {
+	benchmarkProcessorQueueSize(100, b)
+}
+
+func benchmarkProcessor500(b *testing.B) {
+	benchmarkProcessorQueueSize(500, b)
+}
+
+func benchmarkProcessor1000(b *testing.B) {
+	benchmarkProcessorQueueSize(1000, b)
+}
+
+func benchmarkProcessor2000(b *testing.B) {
+	benchmarkProcessorQueueSize(2000, b)
+}
+
+func benchmarkProcessor3000(b *testing.B) {
+	benchmarkProcessorQueueSize(3000, b)
+}
+
+func benchmarkProcessor4000(b *testing.B) {
+	benchmarkProcessorQueueSize(4000, b)
+}
```

Comment:
Given how you are parameterizing within `BenchmarkWithQueueSize` you can get rid of these functions.